table-api/src/main/java/io/deephaven/api/literal/LiteralInt.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.api.literal;
+
+import io.deephaven.annotations.SimpleStyle;
+import org.immutables.value.Value.Check;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+@Immutable
+@SimpleStyle
+abstract class LiteralInt extends LiteralBase {
+
+    public static LiteralInt of(int value) {
+        return ImmutableLiteralInt.of(value);
+    }
+
+    @Parameter
+    public abstract int value();
+
+    @Override
+    public final <T> T walk(Visitor<T> visitor) {
+        return visitor.visit(value());
+    }
+
+    @Check
+    final void checkNotDeephavenNull() {
+        if (value() == Integer.MIN_VALUE) {
```

Comment:
it would probably be better to get an dependency here on QueryConstants, or move those somewhere so that table-api can also reference them.

Reply:
Yep - the dependency was why it wasn't already done. I'll look what it takes to move these constants to somewhere more accessible.